### PR TITLE
feat: Add mongodb v5 support with strict filter types

### PR DESCRIPTION
**Pull request by avaly**
This PR adds support for [`mongodb` v5](https://github.com/mongodb/node-mongodb-native/releases/tag/v5.0.0).

Together with this we're switching to the `StrictFilter` types.

Because this is changing the implementation of the internal types and has potential to break existing applications, I think this should be a major version bump. Happy to hear other arguments though.

- [x] Upgrade `mongodb` to `5.0.0`
- [x] Use `StrictFilter` types
- [x] Use `StrictUpdateFilter` types
- [x] Update MongoDB API docs links

### CODE REVIEW

**Comment by vanstinator** [comment]
Agreed on the major version bump.

**Comment by marafat** [comment]
What is the TL;DR on the difference between `Filter` and `StrictFilter`?

**Comment by avaly** [comment]
> What is the TL;DR on the difference between `Filter` and `StrictFilter`?

`Filter` is the old pre-4.3.0 `Filter` type, which does not support dot-notation fields.

`StrictFilter` is the post-4.3.0 `Filter` type, which does support dot-notation fields.

**Review by marafat — DISMISSED** [review]

**Comment by ejmartin504** [comment]
> Because this is changing the implementation of the internal types and has potential to break existing applications, I think this should be a major version bump. Happy to hear other arguments though.

From looking at the changes, I'm having trouble understanding what this would break in external applications. 

However, since this does involve a major version upgrade of the `mongodb` library, I think that's enough to merit a major version update.

**Comment by ejmartin504** [comment]
Worth pinning to 5.0.1? https://github.com/mongodb/node-mongodb-native/releases/tag/v5.0.1

**Comment by avaly** [comment]
I updated our dev dependency to 5.0.1, but given the peer dependency version is more of an indicator of breaking changes which we depend on, I'll keep that version declared as `>=5.0.0` and consumers can upgrade at their own pace to subsequent v5 versions.

**Review by marafat — DISMISSED** [review]

File: docs/recipes.md [hunk 1]
```diff
@@ -190,3 +190,28 @@ const postSchema = schema(
...
+
+`mongodb` exposes two important TypeScript types for the most common operations you can do with the driver: the `Filter` (used for querying documents) and `UpdateFilter` (used for updating documents) types.
+
+Starting with `mongodb` v4.3.0, these types were enhanced to support dot notation field access and type checking.
+
+However, `mongodb` v5.0.0 [removed these types](https://github.com/mongodb/node-mongodb-native/blob/main/etc/notes/CHANGES_5.0.0.md#dot-notation-typescript-support-removed-by-default) as the default ones used in their methods and reverted to the old ones without dot notation support. The previous enhanced types were not removed, instead they were renamed to `StrictFilter` and `StrictUpdateFilter`, but they aren't referenced in any of their methods.
+
+Papr is using the strict types to provide type safety for all query and update filters.
+
+This comes with a caveat: whenever you need to interact with the `mongodb` driver collections, you need to cast filter types to their simple counterparts, since `Filter` is not compatible with `StrictFilter`.
```
Comment (ejmartin504) [inline docs/recipes.md:204]: This bit of the docs is very nicely written, well done!

**Review by ejmartin504 — APPROVED** [review]

**Pull request merged**